### PR TITLE
Software documentation chapter

### LINKED ISSUE #879: Software documentation chapter
## Detailed description
 * This issue is a place to discuss matters relating to the writing of the chapter on software documentation.
 * This chapter is being written in _pull_request_number_ (tbd)
 * This chapter is being developed on the _chapter_topic_ branch (tbd)

I would like to migrate Documentation chapter from [eScience Center Guide's Documentation Chapter](https://guide.esciencecenter.nl/#/best_practices/documentation)
Documentation touches many things already covered in other parts of the book. For example README or Code of Conduct are described in the collaboration chapter. The Documentation chapter should cover the description of different kinds of documentation and why it is important and link to other places when it makes more sense.

### Table of contents
  1. About
    - Markdown
  2. Project Documentation
    - README
    - CONTRIBUTING (link to collaboration chapter)
    - ROADMAP
    - CHANGELOG
    - LICENSE
    - Code of Conduct (link to collaboration chapter)
    - how to cite this software (expand section in recognition chapter)
  3. Code Documentation
    - code comments
    - API documentation
    - usage documentation
    - development setup
    - description of issue labels
    - code style (link to review)
    - how to file a bug report
    - software version (there should also be somewhere description of semantic versioning)

### Resources
[eScience Center Guide's Documentation Chapter](https://guide.esciencecenter.nl/best_practices/documentation.html)
[Mozilla Open Leadership Training Series](https://mozilla.github.io/open-leadership-training-series/articles/opening-your-project/)

## Current status
This chapter is empty. If anyone would like to collaborate on that, they are welcome.

 - [ ] Write chapter outline
 - [ ] Add material to the chapter
 - [ ] Combine materials into a readable chapter
 - [ ] Proofread
 - [ ] Request reviews
 - [ ] Address reviews
 - [ ] Merge to master branch.

## Updates
no updates yet

### PR DESCRIPTION
**Pull request by c-martinez**
### Summary

This PR adds a new chapter on software documentation as part of the guide on reproducible research.

Fixes #879

### List of changes proposed in this PR (pull-request)

* Summary of the chapter 
* Section on code documentation
* Section on project documentation
* Section for other resources

### What should a reviewer concentrate their feedback on?
- [ ] Everything looks ok?
- [ ] How could this chapter be improved.
- [ ] Are there areas suggested in the issue #879 that are not yet covered

### Acknowledging contributors
- [ ] All contributors to this pull request are already named in the [table of contributors](https://github.com/the-turing-way/the-turing-way/blob/main/README.md#contributors) in the README file.
- [ ] The following people should be added to the [table of contributors](https://github.com/the-turing-way/the-turing-way/blob/main/README.md#contributors) in the README file: 
@manuGil did most of the writing
@svenvanderburg, @paulmaxus and @JaroCamphuijsen contributed to the discussion.

### CODE REVIEW

**Comment by aleesteele** [comment]
@all-contributors please add @manuGil for writing

**Comment by aleesteele** [comment]
@all-contributors please add @manuGil for content

File: book/website/reproducible-research/code-documentation/code-documentation-project.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+The section on [licensing](https://the-turing-way.netlify.app/reproducible-research/licensing) provides excellent detail on open-source licensing.
+
+## Code of Conduct
+
+When collaboration is an important aspect of a software project,
+defining a *code of conduct* is necessary to create and maintain a collaboration environment that promotes participation and fosters the exchage of ideas, 
+while fostering respect among developers. 
+
+The [contributor's covernant](https://www.contributor-covenant.org/) describes a code of conduct for open source software, 
+and this [template](https://github.com/jessesquires/.github/blob/main/CODE_OF_CONDUCT.md) can be used as is or be adapted for a software project.
```
Comment (EstherPlomp) [inline book/website/reproducible-research/code-documentation/code-documentation-project.md:49]:
```suggestion
and this [template](https://github.com/jessesquires/.github/blob/main/CODE_OF_CONDUCT.md) can be used as is or be adapted for a software project.
See also the Turing Way's {ref}`ch-coc`.
```

File: book/website/reproducible-research/code-documentation/code-documentation-resources.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+(rr-documentation-resources)=
+# Other resources
+
+- [The Documentation System](https://docs.divio.com/documentation-system/) A unified theory of documentation.
+- Markdown
```
Comment (EstherPlomp) [inline book/website/reproducible-research/code-documentation/code-documentation-resources.md:5]:
```suggestion
- Markdown
- How to document your research software[](https://coderefinery.github.io/documentation/#how-to-document-your-research-software) by CodeRefinery
```

**Review by EstherPlomp — CHANGES_REQUESTED** [review]
Hi @c-martinez! I think this looks great - I just have some spelling things and internal referencing that need to be addressed and a suggested resource to add :)

File: book/website/reproducible-research/code-documentation/code-documentation-resources.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+(rr-documentation-resources)=
+# Other resources
+
+- [The Documentation System](https://docs.divio.com/documentation-system/): A unified theory of documentation.
+- Markdown
```
Comment (c-martinez) [inline book/website/reproducible-research/code-documentation/code-documentation-resources.md:5]:
```suggestion
- [Markdown](https://daringfireball.net/projects/markdown/)
```

**Comment by c-martinez** [comment]
Thanks @EstherPlomp -- I've incorporated your suggestions, so if you are happy with it, I guess we I can merge it later today :-)

**Review by EstherPlomp — APPROVED** [review]
Fantastic, thanks @c-martinez!

**Pull request merged**